docs/antrea-network-policy.md

```diff
@@ -1221,16 +1320,33 @@ share the `view` ClusterRole to a wider range of subjects to allow them to read
 the policies that may affect their workloads.
 Similar RBAC is applied to the ClusterGroup resource.
 
-## Notes
+## Notes and constraints
 
 - There is a soft limit of 20 on the maximum number of Tier resources that are
   supported. But for optimal performance, it is recommended that the number of
   Tiers in a cluster be less than or equal to 10.
 - In order to reduce the churn in the agent, it is recommended to set the policy
-  priority within the range 1.0 to 100.0.
+  priority (acnp/anp.spec.priority) within the range 1.0 to 100.0.
 - The v1alpha1 policy CRDs support up to 10,000 unique priorities at policy level,
-  and up to 50,000 unique priorities at rule level, across all tiers except for
-  the "baseline" tier. For any two policy rules, their rule level priorities are only
-  considered equal if they share the same tier, and have the same policy priority
-  as well as rule priority.
-- For the "baseline" tier, the max supported unique priorities (at rule level）is 150.
+  and up to 50,000 unique priorities at rule level, across all Tiers except for
+  the "baseline" Tier. For any two Antrea-native policy rules, their rule level
+  priorities are only considered equal if their policy objects share the same Tier
+  and have the same policy priority, plus the rules themselves are of the same rule
+  priority (rule priority is the sequence number of the rule within the policy's
+  ingress or egress section).
+- For the "baseline" Tier, the max supported unique priorities (at rule level) is 150.
+- If there are multiple Antrea-native policy rules created at the same rule-level
+  priority (same policy Tier, policy priority and rule priority), who happen to select
+  overlapping traffic patterns but have conflicting rule actions (e.g.`Allow`v.s.`Deny`),
+  the behavior of such traffic will be undeterministic. For example, consider two
+  AntreaNetworkPolicies created in the same Namespace and Tier with the same policy
+  priority. The first policy applies to all `app=web` Pods in the Namespace and has only
+  one ingress rule to `Deny` all traffic from `role=dev` Pods. The other policy also
+  applies to all `app=web` Pods in the Namespace and has only one ingress rule, which is
+  to `Allow` all traffic from `app=client` Pods. Those two ingress rules might not always
+  conflict, but in case a Pod with both `app=client` and `role=dev` labels initiate
+  traffic towards the web Pods in the Namespace, both rules will be matched at the same
+  priority with conflicting actions. It will be the policy writer's responsibility to
+  identify such ambiguities in rule definitions and avoid potential undeterministic
+  rule enforcement results. In general, it is recommended that rules of conflicting
+  actions are always created at different rule-level priorities.
```

Comment:
honestly, I feel like we should just discourage users from defining policies at the same priority if they apply to the same Pods. Having the notion of rule-level priority across multiple policies is a bit confusing as it is not based on a user-provided priority value, but on the rule position in the list of rules.
what do you think @jianjuns @tnqn ?

Reply:
We can first say the behaviors are not deterministic / predictable if two policies with the same priority apply to the same Pod, if there are both allow and deny rules. Then we can explain further the details like rule priority (assuming most readers will not read it)?

Reply:
I was hoping that with this paragraph it discourages people from defining policies at the same priority if they apply to the same Pods. From Antrea perspective it is hard to enforce this though.